src/OpenFeature.Extensions.Hosting/OpenFeatureBuilder.cs

```diff
@@ -0,0 +1,9 @@
+using Microsoft.Extensions.DependencyInjection;
+
+namespace OpenFeature;
+
+/// <summary>
+/// Describes a <see cref="OpenFeatureBuilder"/> backed by an <see cref="IServiceCollection"/>.
+/// </summary>
+/// <param name="Services"><see cref="IServiceCollection"/></param>
+public sealed record OpenFeatureBuilder(IServiceCollection Services);
```

Comment:
What is the reason to go for a record and the extension methods here? I am not too deep into .NET but I have not seen that pattern formerly.